app/javascript/components/CreateTransaction.jsx

```diff
@@ -0,0 +1,269 @@
+      "end_date": this.state.transaction.end_date,
+      "purchase_date": this.state.transaction.purchase_date,
+      "price": this.state.transaction.price,
+      "buyer_id": 0,
+      "artist_id": artist_id,
+      "work_id": 0,
+      "comment": this.state.transaction.comment
+    }
+
+    if (!valid) {
+      this.setState({ renderErrors: true })
+    } else {
+      Requester.post(transactions_route, payload).then(
+        response => {
+          window.location.href = '/transactions';
+        },
+        error => {
+          console.error(error);
+        }
+      )
+    }
+  }
+
+  componentDidMount = () => {
+    const types_route = APIRoutes.transactions.types;
+    Requester.get(types_route).then(
+      response => {
+        this.setState({
+          types: response,
+          didMount: true
+        });
+      },
+      error => {
+        console.error(error);
+      }
+    )
+  }
+
+  renderPurchaseDate = () => {
+    if (this.state.transaction.transaction_type == "purchase") {
+      return (
+        <div>
+          <p className="f6 lh-copy">Purchase Date  </p>
+          <input
+            type="date"
+            name="purchase_date"
+            id="purchase_date"
+            value={this.state.transaction.purchase_date}
+            onChange={this.handleChange}
+          />
+
+          <p>{this.renderErrorMessages("purchase_date")}</p>
+        </div>
+      )
+    }
+  }
+
+  renderRentalDates = () => {
+    if (this.state.transaction.transaction_type == "rental") {
+      return (
+        <div>
+          <p className="f6 lh-copy">Start Date  </p>
+          <input
+            type="date"
+            name="start_date"
+            id="start_date"
+            value={this.state.transaction.start_date}
+            onChange={this.handleChange}
+          />
+
+          <p>{this.renderErrorMessages("start_date")}</p>
+
+          <p className="f6 lh-copy">End Date  </p>
+          <input
+            type="date"
+            name="end_date"
+            id="end_date"
+            value={this.state.transaction.end_date}
+            onChange={this.handleChange}
+          />
+
+          <p>{this.renderErrorMessages("end_date")}</p>
+        </div>
+      )
+    }
+  }
+
+  renderErrorMessages = (fieldName) => {
+    if (this.state.renderErrors && this.state.formErrors[fieldName].length > 0) {
+      return (
+        <div>
+          <p className="error">{this.state.formErrors[fieldName]}</p>
+        </div>
+      )
+    } else {
+      return (
+        <div />
+      );
+    }
+  }
+
+render() {
+  if (!this.state.didMount) {
+    return (
+      <div />
+    );
+  }
+  return (
+      <div className="create-transaction-form">
+        <h2 className="f4 lh-title submit">Submit Details</h2>
+
+        <p className="f6 lh-copy price">Price  </p>
+        <input
+          type="TEXT"
+          name="price"
+          id="price"
+          value={this.state.transaction.price}
```

Comment:
Bump! We should be styling the price within the component so that there isn't room for error for the user to input values. You can do something like `value={this.currencyDisplay(this.state.transaction.price)}` so that the price is automatically displayed formatted correctly

Reply:
you should be able to find the function online somewhere and you can just copy paste